### PR TITLE
Shared feed fact-checks view

### PR DESCRIPTION
**Pull request by caiosba**
## Description

- New Check Design System components: `<ButtonMain />`, `<ListSort />` and `<Card />` 
- New feed components: `<FeedHeader />`, `<FeedTopBar />` and `<FeedBlankState />`
- New search components: `<SearchResultsCards />` and `<FactCheckCard />`
- Removed component: `<FeedSharingSwitch />`
- Tabs deleted from feed view

Reference: CV2-3209.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Unit tests were implemented for all new components.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/CheckFeedLicenses.js
Comment (caiosba) [inline src/app/CheckFeedLicenses.js:7]: @amoedoamorim do you need anything similar for your code?

Comment (caiosba) [reply src/app/CheckFeedLicenses.js:7]: anyway, I'll wait for you to merge first

File: src/assets/images/empty_feed.png
Comment (brianfleming) [reply src/assets/images/empty_feed.png:1]: ha ha...I was just typing that as a comment @caiosba 
Here is a square svg version with the colors in it, so you can add it using the svg component and resize with font-size.


[empty_feed.svg.zip](https://github.com/meedan/check-web/files/11849497/empty_feed.svg.zip)

Comment (caiosba) [reply src/assets/images/empty_feed.png:1]: Haha... amazing, thank you!

File: src/app/CheckFeedLicenses.js
Comment (amoedoamorim) [reply src/app/CheckFeedLicenses.js:7]: Yes, sounds good!

File: localization/react-intl/src/app/components/feed/FeedBlankState.json
Comment (amoedoamorim) [inline localization/react-intl/src/app/components/feed/FeedBlankState.json:10]: I'm not sure if this works well for translation, but it's helpful to provide more details - such as a fully readable sentence, or possible variable values - in the description when interpolating not so trivial pieces of content like this.

File: localization/react-intl/src/app/components/feed/FeedBlankState.json
Comment (caiosba) [reply localization/react-intl/src/app/components/feed/FeedBlankState.json:15]: Makes sense! Fixed here: https://github.com/meedan/check-web/commit/a38c8ccb35f6e200d54dd90951b92f33b363d123

File: localization/react-intl/src/app/components/feed/FeedBlankState.json
Comment (caiosba) [reply localization/react-intl/src/app/components/feed/FeedBlankState.json:10]: Makes sense! Fixed here: https://github.com/meedan/check-web/commit/a38c8ccb35f6e200d54dd90951b92f33b363d123

File: src/app/components/feed/FeedHeader.js
Comment (brianfleming) [inline src/app/components/feed/FeedHeader.js:7]: Please change this to the settings.svg we have in the icons folder

File: src/app/components/feed/FeedHeader.module.css
Comment (brianfleming) [inline src/app/components/feed/FeedHeader.module.css:25]: please add a font-size of 20px so the icon matches the designs. once the settings icon has been changed to the local svg version it will size correctly

File: src/app/components/feed/FeedTopBar.js
Comment (brianfleming) [inline src/app/components/feed/FeedTopBar.js:32]: please change this to 16px, just to get the icon a bit larger

**Review by brianfleming — CHANGES_REQUESTED** [review]
@caiosba added a few small things, but will continue to review, wanted to raise this question though.
It looks like the filters applied to the list powering the shared feed are always applied to the shared feed itself. This should not be the case. For example...
### My List Filters
Screenshot 2023-06-28 at 9 48 17 AM

### The Shared Feed filters
Screenshot 2023-06-28 at 9 48 52 AM

The shared feed is getting the "inconclusive" filter because I set it on the list. The filters on the shared feed should be independent of the list. The reason is, those filters are to be able to filter the list so I could (in the near future) see fact-checks from other organizations. 
If my organization's list is publishing "inconclusive" fact-checks, while another is publishing to the feed "verified" fact-checks. I need a way to see only  the verified, meaning from the other orgs.

let me know if:
- that makes sense
- if that work is just slated for a different ticket.

File: src/app/components/feed/FeedHeader.js
Comment (amoedoamorim) [inline src/app/components/feed/FeedHeader.js:38]: Perhaps getLicenseName, getLicenseIcon and getLicenseTranslatedName should be moved to CheckFeedLicenses.js?

**Review by amoedoamorim — APPROVED** [review]

**Comment by caiosba** [comment]
@brianfleming , thanks for the comments! I'm going to work on the changes. Regarding the list filters being displayed, I was confused because of the ticket CV2-775. But I agree that in this case it makes sense to hide it, so I'm going to work on it as well.

**Comment by brianfleming** [comment]
> CV2-775

@caiosba CV2-775 is only about the "default" lists like `Suggestions`, `Published`, `Inbox`. In other words the lists that we auto create for people. Those lists have definitions that the user should not be able to change, and currently some of those definitions are either incorrect or not shown in the UI, that is the idea behind CV2-775

On the shared feeds page, the items you see below in the list will be all the items contributed to the feed by the orgs participating (only one org to start). So the filtering on that page is only about filtering that list.

Let me know if you would like to chat about it more!

**Comment by caiosba** [comment]
@brianfleming it all makes sense, thanks!

**Comment by caiosba** [comment]
@brianfleming sorry, I was confused, I thought that the feed page was displaying all filters from the list, but it's only displaying the rating and date fields if the list define them, and the reason for that is that both can't coexist right now and it was already the case for the previous shared feed feature we had. Since we're naming this release the "single org feed", I think we can save it for when we work on the collaborative features.

File: src/app/components/feed/FeedHeader.js
Comment (caiosba) [reply src/app/components/feed/FeedHeader.js]: Yes, good idea, done here: https://github.com/meedan/check-web/commit/d45bfc0253b418cd0293b1bd1267300dde646709

File: src/app/components/feed/FeedTopBar.js
Comment (caiosba) [reply src/app/components/feed/FeedTopBar.js]: Done here: https://github.com/meedan/check-web/commit/d45bfc0253b418cd0293b1bd1267300dde646709

File: src/app/components/feed/FeedHeader.module.css
Comment (caiosba) [reply src/app/components/feed/FeedHeader.module.css:25]: Done here: https://github.com/meedan/check-web/commit/d45bfc0253b418cd0293b1bd1267300dde646709

File: src/app/components/feed/FeedHeader.js
Comment (caiosba) [reply src/app/components/feed/FeedHeader.js]: Done here: https://github.com/meedan/check-web/commit/d45bfc0253b418cd0293b1bd1267300dde646709

**Review by brianfleming — CHANGES_REQUESTED** [review]
looking great @caiosba...two things. One that I think we should fix before merging, and one that maybe can wait.

### To Fix:
Since there is no search on this page, when there is a long name we should allow this feed page header to have the full width. Instead its being cut off:

Screenshot 2023-06-28 at 1 57 47 PM

### Maybe wait
the page background color and container colors need to be update. page vs desgin:

Screenshot 2023-06-28 at 2 03 27 PM

**Comment by caiosba** [comment]
@brianfleming I fixed both, please review :pray:

**Review by brianfleming — APPROVED** [review]
🚀 
infinity

**Pull request merged**
